### DISCUSSION TITLE
1.1.0rc4

### DISCUSSION THREAD

**Discussion by FormerLurker** [FormerLurker/ArcWelderPlugin#217] [General]
At long last a new RC of ArcWelder is available!  I've worked out a bunch of difficult enhancements, and just need some feedback before I push to live.  I'm absolutely committed to getting the final release of 1.1.0 out before the new year.  

Please post any feedback in the [discussion for this release](https://github.com/FormerLurker/ArcWelderPlugin/discussions).

# Enhancements
* Add extrusion rate variance to support Cura's Arachne engine (variable line width).  As an added bonus, this should increase quality if ANY slicer misbehaves by dramatically varying the extrusion rate between moves.
* Add support for travel move conversion.  Now extrusion isn't required for conversion, which should make everyone who is using a pen plotter happy.
* Add workaround for firmware that does not support long gcode commands.  You can now limit the maximum gcode command length.
* Reduce the max radius to reduce the impact of gcodes that are too long for the current firmware implementation.  It now stands at 9999mm.
* Enhance arc generation to convert more G1/G2 commands to arcs when transitioning from retraction/deretraction/travel.  Previously the initial point would not be encoded when switching between extrusion/retraction/travel.
* Enhance segment/travel statistics rendering for python code to include unicode box drawing characters, and add separate statistics for travel move encoding.
* Add a 'Cancel All' button to cancel all pending jobs.
* General enhancements to the tab and settings.
# Bug Fixes
* Settings in the 'Current Run Configuration' are now only updated after saving settings changes.

# Photos of the New Interface

![image](https://user-images.githubusercontent.com/33363128/142780910-4a284c8a-910d-4ff0-b984-60d2ade27755.png)

![image](https://user-images.githubusercontent.com/33363128/142780849-02a56981-3aa4-4d9d-aec5-db8a64f86a19.png)

![image](https://user-images.githubusercontent.com/33363128/142780923-828e5e63-b3e8-45e5-95e9-fb186588490d.png)

![image](https://user-images.githubusercontent.com/33363128/142780947-93b8f9c4-cec8-441f-a26d-1e8740a5fc68.png)

![image](https://user-images.githubusercontent.com/33363128/142780967-3de86ae0-7852-46f8-ad5a-870dbd6b9429.png)

<hr /><em>This discussion was created from the release <a href='https://github.com/FormerLurker/ArcWelderPlugin/releases/tag/1.1.0rc4'>1.1.0rc4</a>.</em>

**Comment by CRCinAU** [comment]
Asking a dumb question - how do I get this into Octoprint? Seems I'm currently running 1.1.0rc2

EDIT: Never mind - I realised the Releases page has the 1.1.0rc4 zip file that can be used in Octoprint... I guess it might just take time for 1.1.0rc4 to appear as an update in the Plugin Manager?

**Reply by CRCinAU** [reply]
Furthermore - it looks like it just rolled out:

![image](https://user-images.githubusercontent.com/3992884/142868290-38d37902-b9f0-44be-9b4f-304123b6e150.png)

**Reply by FormerLurker** [reply]
Awesome!  For anyone else having issues, [check out this discussion](https://github.com/FormerLurker/ArcWelderPlugin/discussions/141) for manually installing the update.

**Comment by nnw522** [comment]
I been looking forward to this update going to load the RC today! Thanks for all your hard work!

**Reply by FormerLurker** [reply]
Thanks!  I'd appreciate any feedback you have!

**Comment by tmillerpdx** [comment]
If you have any particular calibration models, I'd be happy to print them and post the details and photos of the results

**Reply by FormerLurker** [reply]
A benchy is always good.  I'm primarily interested in two things regarding testing:  Are there any errors/typos/etc in the interface, and are the models converted without errors in the gcode.

Things like gyroid infill, archimedean top/bottom file, and wipe (at least in PrusaSlicer) can make the conversion more dramatic.

The new 'Travel Arc' conversion and Cura Arachne support would also be good to try out.  For Arachne, verifying that the variable line width is printing as epxected would be awesome.  ArcWelder now detects changes in the flow rate, and will abort the arc if the difference is too great.  Here are those settings in the interface:

![image](https://user-images.githubusercontent.com/33363128/143604934-41a6ccb9-582e-4bed-9142-f8676cbb7e3c.png)

**Comment by nnw522** [comment]
My suggestion is anything with lots of circular motions, I have found that
duck calls obtained off of thingiverse work pretty well as a stress test
for this. My ender 5 Pro would struggle to print these by default, I mean
painfully slow stuttering. With arc welder, 300 percent increase in print
speed hands down. That's how you can tell the difference.

**Reply by greyhound3** [reply]
I just wanted to state that I made several prints with this release and everything is perfect/super/excellent! (works with up-to-date klipper like caramel, smoother than butter on a hot knife...)

**Comment by CRCinAU** [comment]
Until I got the prompt of a new comment on this topic, I'd completely forgotten I did an update....

I guess that this is a good sign as I haven't seen any issues that made me think about the plugin version change or have to actively do anything.

That makes it a +1 from me.

**Comment by FormerLurker** [comment]
Can I ask if the version number showing up at the bottom of the plugin tab is, in fact, v1.1.0rc4?  I had a strange issue when testing where it was still showing up at 1.1.0rc3, but upon digging in there were two ArcWelder packages in the oprint folder.  The package name was missing the a in arcwelder (rcwelder), so I'm thinking it was a fluke.  Deleting it fixed the issue, but I'm curious if it's some kind of migration issue that I missed.

Also, FYI, I'm going to drop rc5, which has a few fixes for errors that were reported.  It has no new features (other than some additional settings validations that resolved some issues), and I don't expect any problems with it.

**Reply by greyhound3** [reply]
Selecting the arc welder tab shows nice statistics, but I can not find a version number. In Octoprint > Settings-Dialog -> Pluginmanager it shows "Arc-Welder (1.1.0rc4)".

**Reply by FormerLurker** [reply]
Here is where the version should be on the tab:

![image](https://user-images.githubusercontent.com/33363128/144254974-725ac666-8981-46ee-873b-235ceb691317.png)

Mine is slightly newer (hence the +), but yours should say V1.1.0rc4 (hopefully).

**Reply by greyhound3** [reply]
Yes, it does show "Arc Welder v1.1.0rc4" there. I didn't see it, thank you.

**Comment by rerickson1** [comment]
I've been using RC4 heavily for all my prints and its been working flawless.
I always use gyroid infill and have printed many large prints where the objects are primarily round (eggs).
Would love to see an official v1.10 release soon!

**Reply by FormerLurker** [reply]
Thanks @rerickson1!!  I've been back in action recently, and have been working a lot on arc-welder related things.  My first priority is to get the github action to use static linking so the cura plugin works without having to install prerequisites.  My second priority was to get arc enhancements into the prusa firmware (just merged, hooray!).  My third priority is updating the plugin (then Octolapse).  I've made a lot of progress in the last weeks.

**Comment by Yoghoo** [comment]
When are you planning to release a new stable version? It's almost 2 years after 1.0 so think it's time again for a new version. ;-)
